src/formio/components/AddressNL.js

```diff
@@ -0,0 +1,301 @@
+    const initialValue = {...this.emptyValue, ...this.dataValue};
+    const intl = createIntl(this.options.intl);
+
+    this.reactRoot.render(
+      <IntlProvider {...this.options.intl}>
+        <ConfigContext.Provider
+          value={{
+            baseUrl: this.options.baseUrl,
+            requiredFieldsWithAsterisk: this.options.evalContext.requiredFieldsWithAsterisk,
+          }}
+        >
+          <Formik
+            initialValues={initialValue}
+            validationSchema={toFormikValidationSchema(addressNLSchema(required, intl))}
+          >
+            <FormikAddress
+              required={required}
+              formioValues={initialValue}
+              setFormioValues={this.onFormikChange.bind(this)}
+            />
+          </Formik>
+        </ConfigContext.Provider>
+      </IntlProvider>
+    );
+  }
+
+  setValue(value, flags = {}) {
+    const changed = super.setValue(value, flags);
+    // re-render if the value is set, which may be because of existing submission data
+    changed && this.renderReact();
+    return changed;
+  }
+}
+
+const addressNLSchema = (required, intl) => {
+  let postcodeSchema = z.string().regex(/^[1-9][0-9]{3} ?(?!sa|sd|ss|SA|SD|SS)[a-zA-Z]{2}$/);
+  let houseNumberSchema = z.string().regex(/^\d{1,5}$/);
+  if (!required) {
+    postcodeSchema = postcodeSchema.optional();
+    houseNumberSchema = houseNumberSchema.optional();
+  }
+
+  return z
+    .object({
+      postcode: postcodeSchema,
+      houseNumber: houseNumberSchema,
+      houseLetter: z
+        .string()
+        .regex(/^[a-zA-Z]$/)
+        .optional(),
+      houseNumberAddition: z
+        .string()
+        .regex(/^([a-zA-Z0-9]){1,4}$/)
+        .optional(),
+    })
+    .superRefine((val, ctx) => {
+      if (!required) {
+        if (val.postcode && !val.houseNumber) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: intl.formatMessage({
+              descripion:
+                'ZOD error message when AddressNL postcode is provided but not houseNumber',
+              defaultMessage: 'You must provide a house number.',
+            }),
+            path: ['houseNumber'],
+          });
+        }
+
+        if (!val.postcode && val.houseNumber) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: intl.formatMessage({
+              descripion:
+                'ZOD error message when AddressNL houseNumber is provided but not postcode',
+              defaultMessage: 'You must provide a postcode.',
+            }),
+            path: ['postcode'],
+          });
+        }
+      }
+    });
+};
+
+const FormikAddress = ({required, formioValues, setFormioValues}) => {
+  const {values} = useFormikContext();
+
+  useEffect(() => {
+    if (!isEqual(values, formioValues)) {
+      setFormioValues(values);
+    }
+  });
+
+  return (
+    <div className="formio-form">
```

Comment:
```suggestion
    <div className="openforms-form-field-container">
```
`.formio-form` is "reserved" for the top-level. They currently apply the same styling for the spacing, but that's not guaranteed to stay like that.